### DISCUSSION TITLE
Linux border router with native support for PHY?

### DISCUSSION THREAD

**Discussion by mattwood2000** [openthread/openthread#7814] [Q&A] [answered]
Hi, I'm just starting to get acquainted with the Thread architecture.  Is it possible to use Linux's native support for the 802.15.4 MAC/PHY layers to support Thread as a border router?  Everything I've seen talks about a Spinel interface to an NCP or RCP.  But, if my Linux hardware has a radio directly connected via a SPI interface can I just use that instead of the cost and overhead of a separate processor and radio?

Thanks, Matt.

**Comment by jwhui** [comment, marked as answer]
OpenThread does not currently support Linux's native support for IEEE 802.15.4. You would have to add some logic to translate spinel commands into the appropriate SPI commands for the 802.15.4 radio.

Note that the Thread Specification has made some modifications to the IEEE 802.15.4 MAC, which is partially why using the standard IEEE 802.15.4 MAC interface is not sufficient.
